src/main/java/com/axis/system/jenkins/plugins/downstream/cache/BuildCache.java

```diff
@@ -206,9 +212,7 @@ protected void updateCache(Run downstreamRun) {
       List<Run> upstreamBuilds = getUpstreamBuilds(causeAction);
 
       for (Run upstreamBuild : upstreamBuilds) {
-        // We check for identical parents since Rebuilder-plugin defines the retriggered build as
-        // upstream cause, which can lead to some strange side effects in the visualization.
-        if (upstreamBuild == null || upstreamBuild.getParent() == downstreamRun.getParent()) {
```

Comment:
I think it is best to leave this check as it is, there might be other causes which introduces links to the same parent, e.g. this cause looks suspicious https://javadoc.jenkins.io/plugin/promoted-builds/hudson/plugins/promoted_builds/PromotionCause.html